tests/cases/fourslash/convertFunctionToEs6Class12.ts

```diff
@@ -0,0 +1,22 @@
+/// <reference path='fourslash.ts' />
+
+// @allowNonTsExtensions: true
+// @Filename: foo.js
+////function Foo() {
+////    this.a = 0;
+////}
+////Foo[`a b c`] = function () {

```

Comment:
tryGetPropertyName produces a property name for other template strings with substitutions, right? If so, we need a test for that.

Reply:
Do you mean `${x}${y}`?

Reply:
@sandersn `tryGetPropertyName` doesn't handle `TemplateExpression` (``${x}``), only `NoSubstitutionTemplateLiteral` (``). There are several reasons why I omitted `TemplateExpression`:
1. `TemplateExpression` is not a valid property/method name, so need to convert `TemplateExpression` to `StringLiteral` like `["text" + x + "text"]() {}`
2. `TemplateExpression` is not included in `symbol.members` ```(`Foo.prototype[`${1}`] = function () {}```, so there are no members to convert
3. Not sure that using expressions as a property/method name is very common.
4. I added handling `NoSubstitutionTemplateLiteral` because it is part of `StringLike`, it is absolutely valid to use ```Foo.prototype[`a b`] = function () {}``` and get method `static "a b"() {}`.

Does that make sense?

Reply:
@sandersn Can we move forward with these changes?